### ISSUE TITLE
worldmap icon pulsation-on-mouse over missing 

### ISSUE THREAD

**Issue by lynxlynxlynx** [gemrb/gemrb#1244]
Split from #1152 to keep that subviews-only.

On hover, it should pulse to red. This is actually data driven in bg2 (and bg1 only) which has all the color info in the mapicons.bam, including the fade in-out of the red overlay at the end. The previous frames are for individual states (normal, visited and so on).

**Comment by bradallred** [comment]
BG1 is actually missing these frames AFICT, but just using a `ColorAnimation` on it produces good results. just a matter of dialing in the colors.

**Comment by bradallred** [comment]
bah, reverence the wrong issue in c7326e9

**Comment by lynxlynxlynx** [comment]
Migrating the relevant info from the other bug, which is now fixed.

bg1 visited, potential yet unvisited (bluish tint) and unvisited (sepia?):
![image](https://user-images.githubusercontent.com/121515/121421109-1de40380-c96e-11eb-923c-125e2876bc7b.png)

....
The colors are a bit too strong in bg1 and unvisited areas are miscolored (this looks to behave differently than bg2 — only known, border accessible areas unvisited areas are blue), but overall it works:
![image](https://user-images.githubusercontent.com/121515/121537154-f8a1d480-ca03-11eb-8149-f77956f8f552.png)

**Comment by bradallred** [comment]
the uncommitted changes I have look decent. the highlight color here is just pure blue 0x0000ffff
<img width="103" alt="image" src="https://user-images.githubusercontent.com/1126744/121561523-cda08c00-c9d5-11eb-803b-30c6e92af3f0.png">

unvisited+unreachable
<img width="96" alt="image" src="https://user-images.githubusercontent.com/1126744/121562283-89fa5200-c9d6-11eb-8ceb-f4f64bd3159c.png">

unfortunately, we dont have a way to pass render flags to text so thats a problem.

**Comment by lynxlynxlynx** [comment]
Can you make the unvisited one even redder? It looks like you just made it translucent.

**Comment by bradallred** [comment]
<img width="92" alt="image" src="https://user-images.githubusercontent.com/1126744/121577654-fbda9780-c9e6-11eb-8a70-544e35b271e7.png">

how's that?

**Comment by lynxlynxlynx** [comment]
Closer, but is it impossible to make it look like in the original? It was lighter there still (see above).

**Comment by bradallred** [comment]
yes, I am aware. `BLIT_HALF_TRANS` looks perfect (or close enough given the gamma differences), except you can see the roads. Its like they are blending the icon alpha channel with the background color and then doing a half trans blit of the icon

<img width="211" alt="image" src="https://user-images.githubusercontent.com/1126744/121578572-0ba6ab80-c9e8-11eb-9c79-e2d85a83a177.png">

**Comment by lynxlynxlynx** [comment]
What about sepia, too dark?

**Comment by bradallred** [comment]
sepia
<img width="200" alt="image" src="https://user-images.githubusercontent.com/1126744/121580197-b2d81280-c9e9-11eb-931d-5b144b828690.png">

**Comment by lynxlynxlynx** [comment]
The icon palette also has the correct blue color, which is (4,20,164).

**Comment by bradallred** [comment]
the BG1 palette seems to only be greyscale... what icon/palette are you referring to?

**Comment by lynxlynxlynx** [comment]
bg2

**Comment by bradallred** [comment]
I came back to see if any of our new knowledge of blend modes would help us here.

It's clear after thinking about it, that there is almost certainly no blending that can accomplish the unvisited look, especially given how roads are hidden by the icon. This surely must have been accomplished by a palette change. The palettes on the BAMs are 256 color gradients, so I looked though all the MPAL256 palettes, but didn't find any that looked right either (tho some certainly look better). I wonder if there was a hard coded palette.

**Comment by bradallred** [comment]
It's possible that it's one of these palettes that is then combined with either a color mod or sepia. I still havent found any winning combinations, but here is a patch that includes the BG:EE mpal256 (it has more palettes that may have been hard coded before) and allows you to cycle though them on the world map by click the let mouse button to advance and the right to go back.

I've also discovered that our sepia conversion is not the same as the original, so the patch alters that to match the glsl from BG:EE.

[mapicons.patch](https://github.com/gemrb/gemrb/files/11592726/mapicons.patch)

**Comment by lynxlynxlynx** [comment]
bg1 mapicons.bam has a trivial well-ordered greyscale palette, quite appropriate to do programmatic math on. Increasing saturation and changing the hue get you close, but it's still no good enough (same half-trans problem).

A simple way to test if the original did any blending would be to change the worldmap, move one of the icons over the water.

**Comment by lynxlynxlynx** [comment]
I proposed the test since it may help finding a candidate palette, that's all.
